### PR TITLE
feat: move

### LINKED ISSUE #77: Feature: Move song to different position in queue
This should be fairly straight-forward if you've been digging in the code. The goal is for a user to be able to `move 4 5` to swap the song in position `4`  to position `5`. The format is `<from> <to>`.

## Conditions
- If the `from` position is the position of the currently playing song, reply with `You can't move the currently playing song` and return

## Related LOC
Check other commands such as `play` for the validation conditions. The command should be called `move` and have a `m` alias. It should accept two parameters, no optional parameters, and no extra parameters.

### PR DESCRIPTION
**Pull request by 1Kuroyasha**
closes #77

### CODE REVIEW

File: src/controllers/MusicController.ts [hunk 1]
```diff
@@ -297,6 +297,28 @@ export default class MusicController extends Controller {
 		}
 	};
 
+	/**
+	 *
+	 * @param songIndex is the index of the song to be moved.
+	 * @param newIndex is the new index of the song.
+	 */
+	move = async (songIndex: number, newIndex: number) => {
```
Comment (omarabdelaz1z) [inline src/controllers/MusicController.ts:305]: No need to use async.

File: src/commands/move.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+			const { title, url } = controller.queue[songIndex];
+
+			await reply("Command.Move.Moved", textChannel, {
+				id: songIndex + 1,
+				title,
+				url,
+				index: newIndex + 1,
+			});
+
+			await controller.move(songIndex, newIndex);
```
Comment (omarabdelaz1z) [inline src/commands/move.ts]: Remove ```await```

File: src/controllers/MusicController.ts (hunk 1 above)
Comment (omarabdelaz1z) [inline src/controllers/MusicController.ts]: No need to use async.

**Review by omarabdelaz1z — APPROVED** [review]

**Review by omarabdelaz1z — APPROVED** [review]

**Review by omarabdelaz1z — APPROVED** [review]

File: src/commands/move.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+export default class Move extends Command {
+	constructor(client: ValClient) {
+		super(client, {
+			name: "move",
+			category: "Music",
+			cooldown: 5 * 1000,
+			nOfParams: 2,
+			description:
+				"moves a certain track from its position to a specified position",
```
Comment (KL13NT) [inline src/commands/move.ts:15]: `"Moves tracks <from> <to> in the queue. Supports playing a track next when passed 1 parameter only."`

File: src/commands/move.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+			exampleUsage: "4 5",
```
Comment (KL13NT) [inline src/commands/move.ts:16]: `<from> <to>\n4 5\n4`

File: src/controllers/MusicController.ts [hunk 5]
```diff
@@ -297,6 +297,30 @@ export default class MusicController extends Controller {
...
+	/**
+	 *
+	 * @param songIndex is the index of the song to be moved.
+	 * @param newIndex is the new index of the song.
+	 */
+	move = (songIndex: number, newIndex: number) => {
+		let tempSong: Song;
+		const songs = this.queue;
+
+		// eslint-disable-next-line array-callback-return
```
Comment (KL13NT) [inline src/controllers/MusicController.ts:309]: Do not ignore eslint rules, they're there for a reason. Return `false` at the end of the function's body instead.

File: src/controllers/MusicController.ts [hunk 6]
```diff
@@ -297,6 +297,30 @@ export default class MusicController extends Controller {
...
+		const tempQueue = songs.filter(function (song: Song, index: number) {
```
Comment (KL13NT) [inline src/controllers/MusicController.ts:310]: Prefer arrow expressions over function expressions.

File: src/controllers/MusicController.ts [hunk 7]
```diff
@@ -297,6 +297,30 @@ export default class MusicController extends Controller {
...
+			if (index !== songIndex) return song;
```
Comment (KL13NT) [inline src/controllers/MusicController.ts:311]: `return true`.

**Review by KL13NT — APPROVED** [review]

**Pull request merged**